packages/apps/plugins/plugin-map/src/components/MapDetectLocations.ts

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { CollectionType, TableType, type MapType } from '@braneframe/types';
+import { Filter, getSpace, useQuery } from '@dxos/react-client/echo';
+
+import { type Marker } from './MapControl';
+
+/**
+ * Find table rows that are in the same collection as the map.
+ * Return the ones with `latitude` and `longitude` fields,
+ * converted to `Marker`s.
+ *
+ * NOTE: This collects rows from all the collections the map is in,
+ * even if there's more than one. At the time of writing this behavior
+ * won't be exercised since being included in multiple collections
+ * is disabled in the UI.
+ *
+ * If we enable being in more than one collection again we should
+ * reconsider what behavior we actually want in that case.
+ */
+export const useMapDetectLocations = (map: MapType): Marker[] => {
+  const space = getSpace(map);
+  if (!space) {
+    // TODO: not sure if throwing error here is right.
+    // Am I missing a space getter function that always succeeds?
+    throw new Error('MapSection Space is undefined');
```

Comment:
Is `getSpace` always guaranteed to succeed? Would appreciate suggestions on what to do here.

Reply:
The pattern I prefer is just to return undefined. Then the UI can handle that gracefully. Missing data should never result in the user seeing the error boundary component.

Letting things reactively flow like this has benefits. For instance, if you're querying `map`, it will probably get the initial value `undefined` which will trigger this until it resolves, causing a re-flow of reactivity.

Reply:
In this case, you could also return empty list until space has a value.

Reply:
Actually, space can be undefined in `useQuery` for this exact reason. So you can just pass it and let that empty query result flow.

Reply:
Ah, I'm catching on now. Is it best practice to have a no-op option for all custom hooks?

Reply:
You can think of these things as compositional parts of reactive machines which will re-trigger when params and dependencies change. So yeah you often want to have a path for undefined / 'data not ready yet'. When the upstream thing gets data, it should flow through the reactivity graph and trigger updates along the way.

Reply:
Got it, that totally makes sense.